cloudify_cli/tests/commands/test_nodes.py

```diff
@@ -0,0 +1,82 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+"""
+Tests all commands that start with 'cfy nodes'
+"""
+
+from uuid import uuid4
+
+from mock import MagicMock
+from cloudify_rest_client.nodes import Node
+from cloudify_rest_client.node_instances import NodeInstance
+
+from cloudify_cli.tests import cli_runner
+from cloudify_cli.tests.commands.test_cli_command import CliCommandTest
+
+
+class NodesTest(CliCommandTest):
+
+    def setUp(self):
+        super(NodesTest, self).setUp()
+        self._create_cosmo_wd_settings()
+
+    def test_nodes_get(self):
+        self.client.nodes.get = MagicMock(return_value=node_get_mock())
+        self.client.node_instances.list = \
+            MagicMock(return_value=[node_instance_get_mock()])
+        cli_runner.run_cli('cfy nodes get --node-id mongod -d nodecellar')
+
+        with self.assertRaises(SystemExit) as sys_exit:
```

Comment:
I'd separate these additional steps to separate tests - e.g. "test_nodes_get_without_deployment_id" and "test_nodes_get_without_node_id"

generally speaking, in unit tests there's no reason to write long tests - it all runs super fast, and the more tests we have and the shorter they are - the clearer they are and the easier it is to understand what exactly broke when something does break